src/Form/RulesSettingsForm.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Form\RulesSettingsForm.
+ */
+
+namespace Drupal\rules\Form;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Psr\Log\LogLevel;
+
+/**
+ * Provides rules settings form.
+ */
+class RulesSettingsForm extends ConfigFormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'rules_settings_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['rules.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('rules.settings');
+    $form['log'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('System log'),
+    ];
+    $form['log']['log'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Log debug information'),
+      '#default_value' => $config->get('log'),
+    ];
+    $form['log']['log_level_system'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Log level'),
+      '#options' => [
+        LogLevel::WARNING => $this->t('Log all warnings and errors'),
+        LogLevel::ERROR => $this->t('Log errors only'),
+      ],
+      '#default_value' => $config->get('log_level_system') ? $config->get('log_level_system') : LogLevel::WARNING,
+      '#description' => $this->t('Evaluations errors are logged to available loggers.'),
+      '#states' => [
+        // Hide the log_level radios when the debug log is disabled.
+        'invisible' => [
+          'input[name="log"]' => ['checked' => FALSE],
+        ],
+      ],
+    ];
+    $form['debug_screen_log'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Screen log'),
+    ];
+    $form['debug_screen_log']['debug_screen'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Log debug information'),
+      '#default_value' => $config->get('debug_screen'),
+    ];
+    $form['debug_screen_log']['log_level_screen'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Log level'),
+      '#options' => [
+        LogLevel::WARNING => $this->t('Log all warnings and errors'),
+        LogLevel::ERROR => $this->t('Log errors only'),
+      ],
+      '#default_value' => $config->get('log_level_screen') ? $config->get('log_level_screen') : LogLevel::WARNING,
+      '#description' => $this->t('Evaluations errors are logged to available loggers.'),
```

Comment:
Description is wrong. This should be "Level of log messages shown on screen" or similar.